browser/ui/brave_shields_data_controller.cc

```diff
@@ -42,6 +56,23 @@ void BraveShieldsDataController::DidFinishNavigation(
 void BraveShieldsDataController::WebContentsDestroyed() {
   favicon::ContentFaviconDriver::FromWebContents(web_contents())
       ->RemoveObserver(this);
+  observation_.Reset();
+}
+
+void BraveShieldsDataController::OnContentSettingChanged(
+    const ContentSettingsPattern& primary_pattern,
+    const ContentSettingsPattern& secondary_pattern,
+    ContentSettingsTypeSet content_type_set) {
+  // This callback is only used for notifying changes from different tab of
+  // same site.
+  if (settings_changed_by_this_)
+    return;
+
+  if (content_type_set.GetType() == ContentSettingsType::BRAVE_SHIELDS &&
+      primary_pattern.Matches(web_contents()->GetLastCommittedURL())) {
```

Comment:
`web_contents()->GetLastCommittedURL()` this can be replaced by `GetCurrentSiteURL()`? both look similar